util/flags.py

```diff
@@ -42,6 +52,7 @@ def create_flags():
     f.DEFINE_float('augmentation_pitch_and_tempo_scaling_max_pitch', 1.2, 'max value of pitch scaling')
     f.DEFINE_float('augmentation_pitch_and_tempo_scaling_max_tempo', 1.2, 'max vlaue of tempo scaling')
 
+    f.DEFINE_integer('review_audio_steps', 0, 'number of audio, push the audio into summary directory (if 0, no audio will be dumped)')
```

Comment:
1. Yes, one file (randomly picked in batch) per step is saved until the given count is reached, in order not to stuck the prefetch performance.
2. `every X steps` is a good idea, I would like to apply both constraint, because if we use `every X steps` as argument only, user might misuse it like set value to 1, then the folder in tensorboard could be stuffed with audio while training large dataset.

Reply:
What about saving a fixed number of files each epoch? You could get the lenght of the dataset from the length of the training.csv (This may be not the correct lenght as mentioned [here](https://github.com/mozilla/DeepSpeech/issues/2784#issuecomment-590034783), but i think it could be good enough). You could also try [tf.data.experimental.cardinality(dataset)](https://www.tensorflow.org/api_docs/python/tf/data/experimental/cardinality).